### PR TITLE
chore: add semantic release, preset conventionalcommits

### LINKED ISSUE #170: [Infra] Specify engine for node version in package.json
This is a more explicit way to ask for a specific node/npm version than using `.npmrc`, having both would be best

### LINKED ISSUE #180: NPM Auto-publish
### Overview

Merging a branch on master having a new version in package.json it could be great using GH actions to automatically:
- publish a new version on NPM
- create a GH release
- push a git tag

### PR DESCRIPTION
**Pull request by seedy**
## Description

Closes #180 - Adds [semantic-release](https://semantic-release.gitbook.io/semantic-release/)
Closes #170 - Adds node engine requirement following [semantic-release requirement](https://semantic-release.gitbook.io/semantic-release/support/node-version): `>=14.17.0`

Preset: [conventional-commits](https://www.conventionalcommits.org/en/v1.0.0/)

Updates README

Updates package version to `1.0.0-semantic-release` following [recommendations](https://semantic-release.gitbook.io/semantic-release/support/faq#can-i-set-the-initial-release-version-of-my-package-to-0.0.1)

Package version will be handled by `semantic-release`. The one in this project will no longer be updated. We can create an RFC if you think it would be useful to keep updating our package version on github.

## Dependency changes

| Dependency | Version        | State   |
| ---------- | -------------- | ------- |
| [semantic-release](https://github.com/semantic-release/semantic-release)       | ^19.0.2          | Added   |
| [@semantic-release/commit-analyzer](https://github.com/semantic-release/commit-analyzer)       | ^9.0.2          | Added   |
| [@semantic-release/github](https://github.com/semantic-release/github)       | ^8.0.2          | Added   |
| [@semantic-release/npm](https://github.com/semantic-release/npm)       | ^9.0.0          | Added   |
| [@semantic-release/release-notes-generator](https://github.com/semantic-release/release-notes-generator)       | ^10.0.3          | Added   |
| [conventional-changelog-conventionalcommits](https://github.com/conventional-changelog/conventional-changelog)       | ^4.6.3          | Added   |

- [semantic-release](https://semantic-release.gitbook.io/semantic-release) is responsible for auto-releasing PRs, following plugins are the default ones required to handle the full feature:
  - @semantic-release/commit-analyzer
  - @semantic-release/github
  - @semantic-release/npm
  - @semantic-release/release-notes-generator
- [conventional-changelog-conventionalcommits](https://www.conventionalcommits.org/en/v1.0.0/) is the preset we use for the commit analyzer to compute the proper release version, based on commit message merged on master

## Breaking changes

None

## How to test?

You can play with https://github.com/matthieuh/test-semantic-release

## Good PR checkboxes

- [x] Change has been tested
- [ ] Added/Updated tests
- [x] Added/Updated documentation
- [x] PR title follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Labels are set
- [x] Project is linked

## Good Review checkboxes

<details>
<summary> ℹ️  Copy the snippet and paste in the review field to fill it</summary>

```markdown
- [ ] I've tested the changes
- [ ] I've agreed on the unit tests (soon to come)
- [ ] I've checked the stories
- [ ] I've read the code and understood it
- [ ] I don't have any more questions
- [ ] I've described any optional improvements
- [ ] I checked PR title follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
```

</details>

### CODE REVIEW

**Comment by seedy** [comment]
@matthieuh Some questions:
- Do you see any breaking change?
- Do we still require `np` to release?

**Comment by seedy** [comment]
I have been working on a dry run github action POC. 

I'm struggling with manipulating the ouput of semantic-release run...

https://github.com/matthieuh/test-semantic-release/pull/7

There's also a github action which packages semantic-release and computations by itself: 

https://github.com/guilhermetod/semantic-release-notes-preview

**Comment by seedy** [comment]
Tasks left:

- [ ] split dry run and commitlint in spearate PRs
- [ ] prepare the PR for review
- [ ] link to test-semantic-release for testing
- [ ] package.json version `1.0.0-semantic-release`

**Review by paulocfjunior — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.17.0'
```
Comment (paulocfjunior) [inline .github/workflows/release.yml:18]: What about using [LTS](https://github.com/traefik/faency/pull/307/files#diff-9612567e32a0a02130c314154e005de75bf7f2c0e8829eed9fda6e5def99305aR19) here too?

File: package.json [hunk 2]
```diff
@@ -88,5 +94,25 @@
...
+      "next"
+    ],
+    "plugins": [
+      [
+        "@semantic-release/commit-analyzer",
+        {
+          "preset": "conventionalcommits"
+        }
+      ],
+      "@semantic-release/release-notes-generator",
```
Comment (seedy) [inline package.json:110]: I need to check on [test semantic release](https://github.com/matthieuh/test-semantic-release) if the `preset` is needed for release notes generator

**Review by ldez — APPROVED** [review]
LGTM

File: package.json [hunk 3]
```diff
@@ -88,5 +94,30 @@
...
+      ],
+      "@semantic-release/npm",
+      "@semantic-release/github"
+    ]
+  },
+  "engines": {
+    "node": ">=14.17.0"
   }
-}
+}
```
Comment (seedy) [inline package.json:123]: `nth-check` has been downgraded :confused:

**Review by gndz07 — APPROVED** [review]

**Review by matthieuh — APPROVED** [review]

**Pull request merged**